kategory-effects-kotlinx-coroutines/src/main/kotlin/kategory/effects/data/DeferredKW.kt

```diff
@@ -0,0 +1,106 @@
+package kategory.effects
+
+import kategory.*
+import kotlinx.coroutines.experimental.*
+import kotlin.coroutines.experimental.CoroutineContext
+import kotlin.coroutines.experimental.EmptyCoroutineContext
+
+fun <A> Deferred<A>.k(start: CoroutineStart = CoroutineStart.DEFAULT): DeferredKW<A> =
+        DeferredKW(this, start)
+
+fun <A> DeferredKWKind<A>.value(): Deferred<A> = this.ev().deferred
+
+@higherkind
+@deriving(
+        Functor::class,
+        Applicative::class,
+        Monad::class
+)
+data class DeferredKW<out A>(val deferred: Deferred<A>, val start: CoroutineStart = CoroutineStart.DEFAULT) : DeferredKWKind<A>, Deferred<A> by deferred {
```

Comment:
Why does this need to be a data class?  A simple class with an invoke would be more efficient since we don't need to compare structurally DeferredKW instances

Reply:
All datatypes are data classes as a convention, excepts for the ones that are sealed classes + data classes.